### ISSUE TITLE
Tracking Donations made to Giveth through optional donation box

### ISSUE THREAD

**Issue by divine-comedian** [Giveth/analytics-dashboard#15]
When a user is donating to a project on Giveth, they have an option to make a donation to Giveth using a % they want to donate to the project. 

<img width="463" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWorQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--58e38a9b200d8b0f05b9c578ea1495864d99da8c/image.png" alt="image.png" />

We should have a way to track some statistics for our own internal purposes.

Between two dates we should be able to know:

- the number of donations to Giveth made using the optional donation box
- the USD value of donations made to Giveth using optional donation box
- average % from their total donation value that they donated to Giveth.

**Comment by ae2079** [comment]
@divine-comedian 
Do we need to consider recurring donations too?
If yes, some cases are difficult to handle, for example, if a user had a donation stream to giveth from before, when he used the donation box, we just updated the previous stream. thus, we can not retrieve all the data of previous donations using the donation box, because they might be updated so many times.

**Comment by divine-comedian** [comment]
I'm okay to leave recurring donations out of scope.. for now

**Comment by divine-comedian** [comment]
Looks like there is now an error on staging preventing donations from happening associated with this feature development https://github.com/Giveth/analytics-dashboard/issues/9#issuecomment-2218535452

**Comment by ae2079** [comment]
> Looks like there is now an error on staging preventing donations from happening associated with this feature development [#9 (comment)](https://github.com/Giveth/analytics-dashboard/issues/9#issuecomment-2218535452)

yes, I checked it and it related to not executing migration correctly on the staging, it is odd because the pipeline passed.
anyway, I think the problem is related to migration performance because the query for filling data for new columns is very heavy, I will try to break the migration into smaller ones and improve the query performance.

**Comment by divine-comedian** [comment]
@ae2079 - is this resolved now? What is left to develop or is is ready for QA?

**Comment by ae2079** [comment]
> @ae2079 - is this resolved now? What is left to develop or is is ready for QA?

yes, it is resolved on staging and I sent a new PR for the migrations just now to be more efficient for the deployment.

**Comment by ae2079** [comment]
@LatifatAbdullahi @divine-comedian 
you can now test this feature on staging via this link: https://analytics-dashboard-2gg85uoqp-givethio.vercel.app/
Note that the old records are not available until the issue on running migrations fix by DevOps team

**Comment by LatifatAbdullahi** [comment]
Alright!

**Comment by LatifatAbdullahi** [comment]
@divine-comedian @ae2079 

Test status: completed


New Donors Count: Pass
New Donors Donation: Fail


**Donation Box Metrics**

Total Donations to Giveth Using Donation Box: Pass
Total USD Value to Giveth Using Donation Box: Pass
Average Percentage of Donation to Giveth Using Donation Box: Pass

**Comment by ae2079** [comment]
> @divine-comedian @ae2079
> 
> Test status: completed
> 
> * [x]  I do not want want to support Giveth with my donation(checked): Pass
> * [ ]  I do not want want to support Giveth with my donation(unchecked): Pass
> 
> New Donors Count: Pass New Donors Donation: Fail
> 
> **Donation Box Metrics**
> 
> Total Donations to Giveth Using Donation Box: Pass Total USD Value to Giveth Using Donation Box: Pass Average Percentage of Donation to Giveth Using Donation Box: Pass

What did you mean by New Donors Donations?
(I think you mean the section above donation metrics, that is not changed in this branch.)

**Comment by LatifatAbdullahi** [comment]
@ae2079 

Yes the section above the Donation box metrics. Its not included? If yes, then I will just remove it from the comment

**Comment by LatifatAbdullahi** [comment]
@divine-comedian @ae2079 

Test status: completed

- [x] I do not want want to support Giveth with my donation(checked): Pass
- [ ] I do not want want to support Giveth with my donation(unchecked): Pass


**Donation Box Metrics**

Total Donations to Giveth Using Donation Box: Pass
Total USD Value to Giveth Using Donation Box: Pass
Average Percentage of Donation to Giveth Using Donation Box: Pass

**Comment by divine-comedian** [comment]
> @divine-comedian @ae2079
> 
> Test status: completed
> 
> * [x]  I do not want want to support Giveth with my donation(checked): Pass
> * [ ]  I do not want want to support Giveth with my donation(unchecked): Pass
> 
> **Donation Box Metrics**
> 
> Total Donations to Giveth Using Donation Box: Pass Total USD Value to Giveth Using Donation Box: Pass Average Percentage of Donation to Giveth Using Donation Box: Pass

This is for all networks?

**Comment by divine-comedian** [comment]
@ae2079  - I was testing this feature out today and got an error 
![image](https://github.com/user-attachments/assets/b90088b0-05f0-4d4c-9539-39ba66ea8110)

Looks like the stats related are empty, despite being tested by @LatifatAbdullahi - What's going on here?

**Comment by LatifatAbdullahi** [comment]
> @ae2079 - I was testing this feature out today and got an error, using this preview link -> https://analytics-dashboard-2gg85uoqp-givethio.vercel.app/ ![image](https://private-user-images.githubusercontent.com/67759413/348806841-b90088b0-05f0-4d4c-9539-39ba66ea8110.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNjYzODQsIm5iZiI6MTcyMTA2NjA4NCwicGF0aCI6Ii82Nzc1OTQxMy8zNDg4MDY4NDEtYjkwMDg4YjAtMDVmMC00ZDRjLTk1MzktMzliYTY2ZWE4MTEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE1VDE3NTQ0NFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhZGVhOWRlZDRjNzMzZGFjZmVhYTE0NDUyNjhkMzMxMzlmMDA0M2ViZjE4ZDUwOWY2MDZhMDUyMmM0NmNmMjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.g-KGhLTBwpEpXleB3sb2wdcIK3yPrBS-MspCodN_5LQ)
> 
> Looks like the stats related are empty, despite being tested by @LatifatAbdullahi - What's going on here?

This is strange, as it didnt throw this error when I tested and the analytics board worked as expected last week, this is a recent error

**Comment by ae2079** [comment]
> @ae2079 - I was testing this feature out today and got an error, using this preview link -> https://analytics-dashboard-2gg85uoqp-givethio.vercel.app/ ![image](https://private-user-images.githubusercontent.com/67759413/348806841-b90088b0-05f0-4d4c-9539-39ba66ea8110.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNzExODEsIm5iZiI6MTcyMTA3MDg4MSwicGF0aCI6Ii82Nzc1OTQxMy8zNDg4MDY4NDEtYjkwMDg4YjAtMDVmMC00ZDRjLTk1MzktMzliYTY2ZWE4MTEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE1VDE5MTQ0MVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4YjA4ZTJiMWEyY2JjMzhiM2MwMDM5ODI2ZTllODdmN2JhZDBlZThjZDM5NDEwMDk2ZWIyM2Q4YmY4YjQ4NjAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.yX6QqHz-NCbX4013qj5rPfAJAGx3OL9qguXl68p7JKc)
> 
> Looks like the stats related are empty, despite being tested by @LatifatAbdullahi - What's going on here?

@divine-comedian @LatifatAbdullahi 
Please use this link for testing:
https://staging.stats.giveth.io/
this is staging of analytics dashboard, and we will merge all branches to it, so the fixes are apply to it.
and in this case, I merged a PR for getting more related errors, and for this issue, it seems some new changes on the impact graph staging causes this error.
I'll check impact graph changes and fix it ASAP.

**Comment by ae2079** [comment]
There was an issue with running migrations on staging of the impact graph from before and because of that, the migrations of new PR that merged to the staging, did not run, so this error happened.
And today @mohammadranjbarz did some works to fix that so this problem will be fixed after new changes of impact graph deployed on the staging

**Comment by ae2079** [comment]
> @ae2079 - I was testing this feature out today and got an error, using this preview link -> https://analytics-dashboard-2gg85uoqp-givethio.vercel.app/ ![image](https://private-user-images.githubusercontent.com/67759413/348806841-b90088b0-05f0-4d4c-9539-39ba66ea8110.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwNzczMjMsIm5iZiI6MTcyMTA3NzAyMywicGF0aCI6Ii82Nzc1OTQxMy8zNDg4MDY4NDEtYjkwMDg4YjAtMDVmMC00ZDRjLTk1MzktMzliYTY2ZWE4MTEwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MTUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzE1VDIwNTcwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMwZmZiM2NjZjdlNjRlN2Y2NDgwNTY2MzgxZDcwY2FiMjhlYTUzNDgwZWNkNGEyZmM3OWNjNDM0YTE3OTU1YzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6Wjr5ED_zr94KUFhzHjLnaCU8dCPhazhuGfH8WICUmM)
> 
> Looks like the stats related are empty, despite being tested by @LatifatAbdullahi - What's going on here?

Now, this error is fixed, but the request is timeout, and I'm on that to find the problem(maybe the data gathering in the impact graph is taking a long time) when this one is fixed, I'll tell it here

**Comment by ae2079** [comment]
@divine-comedian @LatifatAbdullahi 
now the optimization PR merged on staging and you can test it on that ( https://staging.stats.giveth.io/ )

**Comment by LatifatAbdullahi** [comment]
@divine-comedian @ae2079 

Test status: completed

 I do not want want to support Giveth with my donation(checked): **Pass**
 I do not want want to support Giveth with my donation(unchecked): **Pass**
 
**Donation Box Metrics**

Total Donations to Giveth Using Donation Box: **Pass**
Total USD Value to Giveth Using Donation Box: **Pass**
Average Percentage of Donation to Giveth Using Donation Box: **Pass**

**Comment by ae2079** [comment]
@divine-comedian 
I made the release PRs for this issue:
backend: https://github.com/Giveth/impact-graph/pull/1726
frontend: https://github.com/Giveth/analytics-dashboard/pull/28
and when you approved it, we can release it on production
